app/src/main/java/com/github/se/eventradar/viewmodel/qrCode/ScanTicketQrViewModel.kt

```diff
@@ -0,0 +1,132 @@
+package com.github.se.eventradar.viewmodel.qrCode
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.User
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ScanTicketQrViewModel
+@Inject // Dependency injection
+constructor(
+    private val userRepository: IUserRepository,
+    private val eventRepository: IEventRepository,
+    qrCodeAnalyser: QrCodeAnalyser,
+) : ViewModel() {
+
+  private val _uiState = MutableStateFlow(QrCodeScanTicketState())
+  val uiState: StateFlow<QrCodeScanTicketState> = _uiState
+
+  enum class Action {
+    ScanTicket,
+    ApproveEntry,
+    DenyEntry,
+    FirebaseUpdateError,
+    FirebaseFetchError,
+    AnalyserError,
+  }
+
+  enum class Tab {
+    MyEvent,
+    ScanQr
+  }
+
+  private var myEventID: String? =
+      "1" // TODO MUst be chaNged to NUll and initialized by navigation via saveEventID()
+
+  init {
+    qrCodeAnalyser.onTicketDecoded = { decodedString ->
+      val result = decodedString ?: "Failed to decode QR Code"
+      updateDecodedString(result) // Update state flow
+      if (result != "Failed to decode QR Code") {
+        //        println("correctly decoded")
+        updatePermissions(result) // Directly call updateFriendList
+      } else {
+        //      println("wrongly decoded")
+        changeAction(Action.AnalyserError)
+        //        println("checkpoint 2")
+      }
+    }
+  }
+
+  private fun updatePermissions(decodedString: String) {
+    while (myEventID == null) {
+      // Wait until myEventID is not null
+    }
```

Comment:
Is this needed because of what we've discussed about runtime argument not being able to be passed alongside injected argument in the viewModel constructor ?